### PR TITLE
Feature/vao

### PR DESCRIPTION
**Pull request by matwey**
Check this GUI features before merge:

- [ ] Positive and negative `BITPIX` images
- [x] Zoom, rotate and flip
- [x] Level control
- [x] Color maps
- [x] Open new image in the same window and in new window

### CODE REVIEW

File: src/openglfeatures.cpp [hunk 1]
```diff
@@ -29,6 +30,12 @@ bool OpenGLFeatures::checkHasOpenGLHelper(const QOpenGLContext& opengl_context,
...
 
+bool OpenGLFeatures::checkHasExtensionHelper(const QOpenGLContext& opengl_context, const char* extension, const quint64 mask) {
+	bool has = opengl_context.hasExtension(extension);
+	bitmask_ |= (has ? mask : static_cast<quint64>(0));
+	return has;
+}
+
 bool OpenGLFeatures::checkHasOpenGL33(const QOpenGLContext& opengl_context) {
 	return checkHasOpenGLHelper(opengl_context, 3, 3,
 		mask_hasOpenGL33 | mask_hasOpenGL30 | mask_hasOpenGL21);
```
Comment (hombit) [inline src/openglfeatures.cpp]: Here we should use only `mask_hasOpenGL33` because of shader incompatibility

File: src/openglfeatures.cpp [hunk 2]
```diff
@@ -42,8 +49,13 @@ bool OpenGLFeatures::checkHasOpenGL21(const QOpenGLContext& opengl_context) {
 	return checkHasOpenGLHelper(opengl_context, 2, 1, mask_hasOpenGL21);
 }
 
-OpenGLFeatures::OpenGLFeatures(const QOpenGLContext& opengl_context) {
+bool OpenGLFeatures::checkHasARB_vertex_array_object(const QOpenGLContext& opengl_context) {
```
Comment (hombit) [inline src/openglfeatures.cpp]: I haven't this extension on MacOS with OpenGL 2.1 Core. Can we just try to use VAO without the support it in extension, Qt allows to check if VAO is supported:
http://doc.qt.io/Qt-5/qopenglvertexarrayobject.html#create

File: src/openglfeatures.cpp (hunk 1 above)
Comment (matwey) [reply src/openglfeatures.cpp]: Thank you, I will do separate PR with fix for 04655711959b69c08fcecf64b07f971549738716

File: src/openglfeatures.cpp (hunk 2 above)
Comment (matwey) [reply src/openglfeatures.cpp]: You are right, it seems to be possible not to use VAO (but still use VBOs)

**Pull request merged**